github-actions/scan/src/input-helper.ts

```diff
@@ -16,3 +19,62 @@ export function split(input: string): string[] {
         .map(item => item.trim())
         .filter(item => item.length > 0);
 }
+
+/**
+ * This method checks the default environment variables 'http_proxy' and 'https_proxy' for http/https proxy specification.
+ * The variables are expected to be specified in this format: http_proxy='http://user:password@proxy.example.org:1234'.
+ * This results in an URL object like this:
+ * URL {
+      href: 'http://user:password@proxy.example.org:1234/',
+      origin: 'http://proxy.example.org:1234',
+      protocol: 'http:',
+      username: 'user',
+      password: 'password',
+      host: 'proxy.example.org:1234',
+      hostname: 'proxy.example.org',
+      port: '1234',
+      pathname: '/',
+      search: '',
+      searchParams: URLSearchParams {},
+      hash: ''
+    }
+ * @returns configured AxiosProxyConfig or undefined if no proxy was
+ * @throws error if the proxy URL inside the ENV variable is an invalid URL
+ */
+export function resolveProxyConfig(): AxiosProxyConfig | undefined {
+    const httpsProxy = process.env.https_proxy || undefined;
+    const httpProxy = process.env.http_proxy || undefined;
+
+    const proxy = httpsProxy || httpProxy;
+    
+    if (!proxy) {
+        return undefined;
+    }
+    try {
+        const proxyUrl = new URL(proxy);
+        const proxyConfig = {
+            protocol: proxyUrl.protocol.replace(':', ''), // Remove the trailing colon
+            host: proxyUrl.hostname,
+            port: proxyUrl.port ? parseInt(proxyUrl.port, 10) : getProtocolDefaultPort(proxyUrl.protocol),
+            ...(proxyUrl.username || proxyUrl.password ? {
+                auth: {
+                    username: proxyUrl.username,
+                    password: proxyUrl.password
+                }
+            } : undefined)
+        };
+        return proxyConfig;
+    } catch(error: any) {
+        throw new Error(`Trying to setup proxy configuration received the error: "${error.message}". Make sure to use the following syntax: http://user:password@proxy.example.org:1234 or without credentials: http://proxy.example.org:1234`);
+    }
+}
+
+function getProtocolDefaultPort(protocol: string) {
+    if (protocol === 'http:') {
+        return 80;
+    } else if (protocol === 'https:') {
+        return 443;
+    } else {
+        throw new Error('Accepted protocols are "http" and "https"');
+    }
```

Comment:
Hmm... you rely on the assumption that the protocol will always contain the colon at the end. Maybe this would be easier/more resilient:

```suggestion
    if (!protocol){
     throw new Error('No protocol defined!');
    }
    if (protocol.startsWith('https') {
        return 443;
    } else if (protocol.startsWith('http') {
        return 80;
    } else {
        throw new Error('Accepted protocols are "http" and "https"');
    }
```